### PR TITLE
PB-442: change timeslider css for partial data

### PR DESCRIPTION
**Pull request by ltkum**
We changed the color of the time slider bar when there is only partial data.
We also added a tippy on hovering the time slider bar, telling us which color is used for which state (no data, partial data, full data)
We also added a small correction to a german translation where a capitalized i shouldn't have been capitalized.

[Test link with data](https://sys-s.dev.bgdi.ch/xz5d234r28tw)

[Test link](https://sys-map.dev.bgdi.ch/preview/feat-pb-442-css-change-timeslider-partial-data/index.html)

### CODE REVIEW

**Review by ltshb — COMMENTED** [review]
Perso I would make the tippy simpler as currently 
image
I would use kind of legend way having a list of square with the color and next to it the meaning.

File: src/modules/i18n/locales/en.json [hunk 1]
```diff
@@ -484,6 +484,10 @@
     "time_hide": "Disable representation of data time stamps.",
     "time_select_year": "Select a year",
     "time_show": "Enable representation of data time stamps.",
+    "time_slider_legend_tippy_intro": "The time slider displays different colours depending on whether the time layers have data for the selected year or not.",
```
Comment (ltshb) [inline src/modules/i18n/locales/en.json:487]: In the UI we never talk about `layer` but about `map` so we should keep this nomenclature in help text.

```
The time slider displays different colors depending on whether the maps have data for the year or not.
```

For German and Italian, @hansmannj and @Luke252 can maybe help and also check english 😉 . For french a proposal that can be double check by you and @pakb 

```
Le curseur de temps affiche différentes couleurs selon si les cartes temporelles disposent de données ou non.
```

File: src/modules/map/components/toolbox/TimeSlider.vue [hunk 2]
```diff
@@ -172,14 +179,22 @@ onMounted(() => {
...
         hideOnClick: false,
         placement: 'bottom',
         trigger: 'manual',
         theme: 'danger',
     })
+    tippyTimeSliderInfo = tippy(timeSliderBar.value, {
+        content: `<div>${i18n.t('time_slider_legend_tippy_intro')}</div><div><div class="color-tippy-data-none"></div><div>${i18n.t('time_slider_legend_tippy_no_data')}</div></div><div><div class="color-tippy-data-partial"></div><div>${i18n.t('time_slider_legend_tippy_partial_data')}</div></div><div><div class="color-tippy-data-full"></div><div>${i18n.t('time_slider_legend_tippy_full_data')}</div></div>`,
+        hideOnClick: true,
+        placement: 'bottom',
+        theme: 'light-border',
```
Comment (ltshb) [inline src/modules/map/components/toolbox/TimeSlider.vue:194]: I would not set the theme here and let the default.

File: src/modules/map/components/toolbox/TimeSlider.vue [hunk 3]
```diff
@@ -172,14 +179,22 @@ onMounted(() => {
...
-        content: tooltipContent,
+    tippyYearOutsideRange = tippy(yearCursorInput.value, {
+        content: `${i18n.t('outside_valid_year_range')} ${ALL_YEARS[0]}-${ALL_YEARS[ALL_YEARS.length - 1]}`,
         arrow: true,
         hideOnClick: false,
         placement: 'bottom',
         trigger: 'manual',
         theme: 'danger',
     })
+    tippyTimeSliderInfo = tippy(timeSliderBar.value, {
```
Comment (ltshb) [inline src/modules/map/components/toolbox/TimeSlider.vue:190]: For this tippy maybe using the [followCursor](https://atomiks.github.io/tippyjs/v6/all-props/#followcursor) with `followCursor: 'horizontal',` might provide a better user experience ?

File: src/modules/i18n/locales/it.json [hunk 4]
```diff
@@ -484,6 +484,10 @@
     "time_hide": "Disattivare la visualizzazione storica dei dati",
     "time_select_year": "Scegliete un anno",
```
Comment (Luke252) [inline src/modules/i18n/locales/it.json:485]: "Scelga" un anno (Scegliere = Infinitif)

File: src/modules/i18n/locales/de.json [hunk 5]
```diff
@@ -484,6 +484,10 @@
     "time_hide": "Deaktivieren der Anzeige von Daten-Zeitständen.",
     "time_select_year": "Wählen Sie ein Jahr aus",
     "time_show": "Aktivieren der Anzeige von Daten-Zeitständen.",
+    "time_slider_legend_tippy_intro": "Der Zeitschieber zeigt verschiedene Farben an, je nachdem, ob die Zeitschichten Daten für das ausgewählte Jahr haben oder nicht.",
```
Comment (Luke252) [inline src/modules/i18n/locales/de.json:487]: "haben" durch "enthalten" ersetzten ?

File: src/modules/i18n/locales/it.json [hunk 6]
```diff
@@ -484,6 +484,10 @@
...
     "time_show": "Attivare la visualizzazione storica dei dati",
+    "time_slider_legend_tippy_intro": "Il cursore del tempo mostra colori diversi a seconda che gli strati temporali abbiano o meno dati per l'anno selezionato.",
```
Comment (Luke252) [inline src/modules/i18n/locales/it.json:487]: "meno" durch "no" ersetzen

File: src/modules/i18n/locales/it.json [hunk 7]
```diff
@@ -484,6 +484,10 @@
...
+    "time_slider_legend_tippy_full_data": "Vollständige Daten",
```
Comment (Luke252) [inline src/modules/i18n/locales/it.json:488]: Dati completi

File: src/modules/i18n/locales/it.json [hunk 8]
```diff
@@ -664,5 +668,19 @@
     "share_map_title": "Link al geoportale della Confederazione Svizzera",
     "link_description": "Descrizione del link  (facoltativo)",
     "media_disclaimer": "Contiene contenuti di terze parti",
-    "drawing_attached": "Disegno aggiunto come allegato"
+    "drawing_attached": "Disegno aggiunto come allegato",
+    "switch_tooltip_attached_to_feature": "Collegare il tooltip alla geometria",
+    "switch_tooltip_floating": "Utilizzare un tooltip fluttuante",
+    "field_required": "Questo campo è obbligatorio",
+    "invalid_email": "e-mail non valide",
```
Comment (Luke252) [inline src/modules/i18n/locales/it.json:675]: e-mail non "valido"

File: src/modules/i18n/locales/it.json [hunk 9]
```diff
@@ -664,5 +668,19 @@
...
+    "no_email": "Il campo email è obbligatorio",
+    "file_unsupported_format": "Questo formato file non è supportato. Sono ammessi solo i seguenti formati: ALLOWED_FORMATS",
+    "form_invalid": "Il modulo non è valido",
+    "file_imported_success": "File importato con successo",
+    "3d_labels_active": "Rimuovi nomi geografici",
```
Comment (Luke252) [inline src/modules/i18n/locales/it.json:680]: "Rimuovere" nomi geografici

File: src/modules/i18n/locales/it.json [hunk 10]
```diff
@@ -664,5 +668,19 @@
...
+    "3d_labels_inactive": "Aggiungi nomi geografici",
+    "3d_vegetation_active": "Rimuovi vegetazione",
```
Comment (Luke252) [inline src/modules/i18n/locales/it.json:682]: "Rimuovere" vegetazione

File: src/modules/i18n/locales/it.json [hunk 11]
```diff
@@ -664,5 +668,19 @@
...
+    "switch_tooltip_attached_to_feature": "Collegare il tooltip alla geometria",
+    "switch_tooltip_floating": "Utilizzare un tooltip fluttuante",
+    "field_required": "Questo campo è obbligatorio",
+    "invalid_email": "e-mail non valide",
+    "no_email": "Il campo email è obbligatorio",
+    "file_unsupported_format": "Questo formato file non è supportato. Sono ammessi solo i seguenti formati: ALLOWED_FORMATS",
+    "form_invalid": "Il modulo non è valido",
+    "file_imported_success": "File importato con successo",
+    "3d_labels_active": "Rimuovi nomi geografici",
+    "3d_labels_inactive": "Aggiungi nomi geografici",
```
Comment (Luke252) [inline src/modules/i18n/locales/it.json:681]: "Aggiungere" nomi geografici

File: src/modules/i18n/locales/it.json [hunk 12]
```diff
@@ -664,5 +668,19 @@
...
+    "3d_vegetation_active": "Rimuovi vegetazione",
+    "3d_vegetation_inactive": "Aggiungi vegetazione",
```
Comment (Luke252) [inline src/modules/i18n/locales/it.json:683]: "Aggiungere" vegetazione

File: src/modules/i18n/locales/en.json [hunk 13]
```diff
@@ -484,6 +484,10 @@
...
+    "time_slider_legend_tippy_full_data": "Full Data.",
```
Comment (Luke252) [inline src/modules/i18n/locales/en.json:488]: "Complete" data ODER "All" data

File: src/modules/i18n/locales/en.json [hunk 14]
```diff
@@ -664,5 +668,19 @@
     "share_map_title": "Link to the geoportal of Swiss Confederation",
```
Comment (Luke252) [inline src/modules/i18n/locales/en.json:668]: of "the" Swiss Confederation

File: src/modules/i18n/locales/en.json [hunk 15]
```diff
@@ -664,5 +668,19 @@
...
     "link_description": "Link description (optional)",
     "media_disclaimer": "Contains third party content",
-    "drawing_attached": "Drawing added as attachment"
+    "drawing_attached": "Drawing added as attachment",
+    "switch_tooltip_attached_to_feature": "Attach tooltip to geometry",
+    "switch_tooltip_floating": "Use a floating tooltip",
+    "field_required": "This field is required",
+    "invalid_email": "Invalid email",
+    "no_email": "Email field is required",
```
Comment (Luke252) [inline src/modules/i18n/locales/en.json:676]: Folgend den anderen Übersetzungen -> "The" e-mail field is required

File: src/modules/i18n/locales/en.json [hunk 16]
```diff
@@ -664,5 +668,19 @@
...
+    "file_unsupported_format": "This file format is not supported.  Only the following formats are allowed: ALLOWED_FORMATS",
+    "form_invalid": "Form is invalid",
+    "file_imported_success": "File successfully imported",
+    "3d_labels_active": "Remove geographic names",
+    "3d_labels_inactive": "Add geographic names",
+    "3d_vegetation_active": "Remove vegetations",
```
Comment (Luke252) [inline src/modules/i18n/locales/en.json:682]: Proposal -> "Remove (and ADD) vegetation" without "s", as vegetation is also used uncountable

File: src/modules/i18n/locales/en.json (hunk 1 above)
Comment (Luke252) [reply src/modules/i18n/locales/en.json:487]: @ltshb I checked and compared the IT and EN versions with the rest. Hope it's clear ;)

File: src/modules/i18n/locales/de.json (hunk 5 above)
Comment (hansmannj) [reply src/modules/i18n/locales/de.json]: The word "Zeitschichten" is strange. I'd use "ob die dargestellten Layer Daten für das ausgewählte ..."
We use "Layer" in other places for the German texts, and that seems OK to me.

File: src/modules/i18n/locales/de.json [hunk 17]
```diff
@@ -664,5 +668,19 @@
     "share_map_title": "Link zum Geoportal der Schweizerischen Eidgenossenschaft",
     "link_description": "Link Beschreibung (optional)",
     "media_disclaimer": "Enthält Inhalte von Drittanbietern",
-    "drawing_attached": "Zeichnung als Anhang hinzugefügt"
+    "drawing_attached": "Zeichnung als Anhang hinzugefügt",
+    "switch_tooltip_attached_to_feature": "Tooltip an Geometrie anhängen",
+    "switch_tooltip_floating": "Eine schwebende Tooltip verwenden",
```
Comment (hansmannj) [inline src/modules/i18n/locales/de.json:673]: Eine*n* schwebende*n*

File: src/modules/i18n/locales/de.json [hunk 18]
```diff
@@ -664,5 +668,19 @@
...
+    "field_required": "Dieses Feld ist erforderlich",
+    "invalid_email": "ungültige E-Mail",
+    "no_email": "Das E-Mail-Feld ist erforderlich",
+    "file_unsupported_format": "Dieses Dateiformat wird nicht unterstützt. Nur die folgenden Formate sind erlaubt: ALLOWED_FORMATS",
+    "form_invalid": "Das Formular ist ungültig",
+    "file_imported_success": "Datei erfolgreich importiert",
+    "3d_labels_active": "Entferne geografische Namen",
+    "3d_labels_inactive": "Geografische Namen hinzufügen",
+    "3d_vegetation_active": "Entferne Vegetationen",
+    "3d_vegetation_inactive": "Vegetationen hinzufügen",
```
Comment (hansmannj) [inline src/modules/i18n/locales/de.json]: I'd use the singular "Vegetation" here in both cases

File: src/modules/i18n/locales/de.json (hunk 5 above)
Comment (hansmannj) [reply src/modules/i18n/locales/de.json]: OK, just saw that for EN we use map instead of layer, so it would be "ob die dargestellten Karten für das ..."

**Comment by hansmannj** [comment]
Most of my comments are gone, I'll try to remember and re-write them :wink:

File: src/modules/i18n/locales/de.json [hunk 19]
```diff
@@ -484,7 +484,7 @@
     "time_hide": "Deaktivieren der Anzeige von Daten-Zeitständen.",
     "time_select_year": "Wählen Sie ein Jahr aus",
     "time_show": "Aktivieren der Anzeige von Daten-Zeitständen.",
-    "time_slider_legend_tippy_intro": "Der Zeitschieber zeigt verschiedene Farben an, je nachdem, ob die Zeitschichten Daten für das ausgewählte Jahr haben oder nicht.",
+    "time_slider_legend_tippy_intro": "Der Zeitschieber zeigt verschiedene Farben an, je nachdem, ob die Karte Daten für das ausgewählte Jahr enthalten oder nicht.",
```
Comment (hansmannj) [inline src/modules/i18n/locales/de.json:487]: As in EN, we should use the plural here, "Karte*n*"

File: src/modules/i18n/locales/en.json [hunk 20]
```diff
@@ -484,8 +484,8 @@
     "time_hide": "Disable representation of data time stamps.",
     "time_select_year": "Select a year",
     "time_show": "Enable representation of data time stamps.",
-    "time_slider_legend_tippy_intro": "The time slider displays different colours depending on whether the time layers have data for the selected year or not.",
-    "time_slider_legend_tippy_full_data": "Full Data.",
+    "time_slider_legend_tippy_intro": "The time slider displays different colors depending on whether the maps have data for the year or not.",
+    "time_slider_legend_tippy_full_data": "Complete data",
     "time_slider_legend_tippy_no_data": "No data.",
     "time_slider_legend_tippy_partial_data": "Partial data.",
```
Comment (hansmannj) [inline src/modules/i18n/locales/en.json:490]: Maybe remove the "." at the end, as in the other texts and languages, too?

File: src/modules/i18n/locales/de.json (hunk 19 above)
Comment (ltkum) [reply src/modules/i18n/locales/de.json]: indeed

**Review by ltshb — APPROVED** [review]
I allowed myself to do some improvement on the tippy and rebased.

**Pull request merged**